apis/identity/v1alpha1/iamusergroupattachment_types.go

```diff
@@ -0,0 +1,93 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// IAMUserGroupAttachmentParameters define the desired state of an AWS IAMUserGroupAttachment.
+type IAMUserGroupAttachmentParameters struct {
+
+	// GroupName is the Amazon IAM Group Name (IAMGroup) of the IAM group you want to
+	// add User to.
+	// +immutable
+	GroupName string `json:"groupName"`
+
+	// UserName presents the name of the IAMUser.
+	// +optional
+	UserName *string `json:"userName,omitempty"`
+
+	// UserNameRef references to an IAMUser to retrieve its userName
+	// +optional
+	UserNameRef *runtimev1alpha1.Reference `json:"userNameRef,omitempty"`
+
+	// UserNameSelector selects a reference to an IAMUser to retrieve its userName
+	// +optional
+	UserNameSelector *runtimev1alpha1.Selector `json:"userNameSelector,omitempty"`
+}
+
+// An IAMUserGroupAttachmentSpec defines the desired state of an
+// IAMUserGroupAttachment.
+type IAMUserGroupAttachmentSpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  IAMUserGroupAttachmentParameters `json:"forProvider"`
+}
+
+// IAMUserGroupAttachmentObservation keeps the state for the external resource
+type IAMUserGroupAttachmentObservation struct {
+	// AttachedGroupARN is the arn for the attached group. If nil, the group
+	// is not yet attached
+	AttachedGroupARN string `json:"attachedGroupArn"`
+}
+
+// An IAMUserGroupAttachmentStatus represents the observed state of an
+// IAMUserGroupAttachment.
+type IAMUserGroupAttachmentStatus struct {
+	runtimev1alpha1.ResourceStatus `json:",inline"`
+	AtProvider                     IAMUserGroupAttachmentObservation `json:"atProvider"`
+}
+
+// +kubebuilder:object:root=true
+
+// An IAMUserGroupAttachment is a managed resource that represents an AWS IAM
+// User group attachment.
+// +kubebuilder:printcolumn:name="USERNAME",type="string",JSONPath=".spec.forProvider.userName"
+// +kubebuilder:printcolumn:name="GROUPNAME",type="string",JSONPath=".spec.forProvider.groupName"
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+type IAMUserGroupAttachment struct {
```

Comment:
Would `IAMGroupUserMembership` be a good name?

Reply:
Yup, seems good and pushed the changes